### ISSUE TITLE
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType

### ISSUE THREAD

**Issue by TheEdgeOfRage** [tiran/certifi-system-store#15]
TL;DR for future reference: There was a symlink for an old version (`/usr/lib/python3.9/site-packages/certifi-3021.3.13-py3.9.egg-info` in my case, though your exact version may differ), that was interfering when running the patch process using the new code. After removing that old symlink and rerunning `python -m certifi -v` it ran just fine.

---

I've written a PKGBUILD file to build the `python-certifi-system-store` package for Archlinux, but it seems to break for the latest release. 

When I run 
```
python -m certifi -v
```

I get the following error

```
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.9/site-packages/certifi/__main__.py", line 14, in <module>
    _patched, _css_dist_info, _certifi_dist_info = _patch_dist_info()
  File "/usr/lib/python3.9/site-packages/certifi/_patch.py", line 31, in _patch_dist_info
    if os.path.samefile(css_dist.egg_info, certifi_dist.egg_info):
  File "/usr/lib/python3.9/genericpath.py", line 101, in samefile
    s2 = os.stat(f2)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
```

The `certifi_dist` object returns None for the `egg_info` property since the package that I've built only generates the `egg_info` for the `certifi_system_store` package, not for `certifi`.

Here's the whole PKGBUILD that I use:

```
pkgbase='python-certifi-system-store'
pkgname=('python-certifi-system-store')
_module='certifi-system-store'
pkgver='3021.3.16'
pkgrel=1

pkgdesc="A certifi hack to use system trust store on Linux/FreeBSD"
url="https://github.com/tiran/certifi-system-store"
license=('MPL')
arch=('any')

depends=('python')
makedepends=('python-setuptools')
provides=('python-certifi')
conflicts=('python-certifi')

source=("https://files.pythonhosted.org/packages/source/${_module::1}/$_module/$_module-$pkgver.tar.gz")
sha256sums=('603be2b260ae2d5d025f584a219087683ff2ddcd09325ebb56b172cd07877057')

build() {
    cd "${srcdir}/${_module}-${pkgver}"
    python setup.py build
}

package() {
    depends+=()
    cd "${srcdir}/${_module}-${pkgver}"
    python setup.py install --root="${pkgdir}" --optimize=1 --skip-build
}
```

I'm not sure how familiar you are with the PKGBUILD format, but the gist of it is that it runs `python setup.py build` to generate all the required files and then `python setup.py install ...`  to copy all the output files into the package dir from which it builds the pacman package.

Here's the list of files that the build process puts in the output package:

```
/usr/lib/python3.9/site-packages/certifi/cacert.pem
/usr/lib/python3.9/site-packages/certifi/core.py
/usr/lib/python3.9/site-packages/certifi/__init__.py
/usr/lib/python3.9/site-packages/certifi/__main__.py
/usr/lib/python3.9/site-packages/certifi/_patch.py
/usr/lib/python3.9/site-packages/certifi/__pycache__/core.cpython-39.opt-1.pyc
/usr/lib/python3.9/site-packages/certifi/__pycache__/core.cpython-39.pyc
/usr/lib/python3.9/site-packages/certifi/__pycache__/__init__.cpython-39.opt-1.pyc
/usr/lib/python3.9/site-packages/certifi/__pycache__/__init__.cpython-39.pyc
/usr/lib/python3.9/site-packages/certifi/__pycache__/__main__.cpython-39.opt-1.pyc
/usr/lib/python3.9/site-packages/certifi/__pycache__/__main__.cpython-39.pyc
/usr/lib/python3.9/site-packages/certifi/__pycache__/_patch.cpython-39.opt-1.pyc
/usr/lib/python3.9/site-packages/certifi/__pycache__/_patch.cpython-39.pyc
/usr/lib/python3.9/site-packages/certifi_system_store-3021.3.16-py3.9.egg-info/dependency_links.txt
/usr/lib/python3.9/site-packages/certifi_system_store-3021.3.16-py3.9.egg-info/PKG-INFO
/usr/lib/python3.9/site-packages/certifi_system_store-3021.3.16-py3.9.egg-info/requires.txt
/usr/lib/python3.9/site-packages/certifi_system_store-3021.3.16-py3.9.egg-info/SOURCES.txt
/usr/lib/python3.9/site-packages/certifi_system_store-3021.3.16-py3.9.egg-info/top_level.txt
/usr/lib/python3.9/site-packages/certifi_system_store-3021.3.16-py3.9.egg-info/zip-safe
```

I presume that my package build process is wrong, though I'm not sure what else I need to run to generate the `certifi.egg_info` data as well.

**Comment by tiran** [comment]
Thanks for trying the packge on Arch! Could you try the version from git HEAD? I changed the verification code a bit but haven't released a new version yet.

By the way you have to build and package both egg-info / dist-info directories. The second directory ensures that pip, setuptools, and pkg-resources think that certifi is installed.

**Comment by TheEdgeOfRage** [comment]
After building the new package I got a different (though similar error), but after some time realised that I had an old symlink in my system's site-packages for this package. More specifically, this one:

```
/usr/lib/python3.9/site-packages/certifi-3021.3.13-py3.9.egg-info -> certifi_system_store-3021.3.13-py3.9.egg-info
```

After removing that file, `python -m certifi -v` ran just fine, so I though that it might have been the problem all along. I was correct, since after building the latest release from GitHub and installing it, it worked this time.

Now, I don't know whether I still need to figure out a way to clean up old versions of the symlinks when upgrading the package or not? I noticed that the old version linked to egg-info, while the new versions use dist-info, so I'm not sure whether that was the problem, or there isn't any cleanup at all?
